replay_server/db_conn.py

```diff
@@ -0,0 +1,52 @@
+import asyncio
+from typing import List
+
+import aiomysql
+from aiomysql import create_pool, Pool
+
+from replay_server.logger import logger
+from replay_server.constants import MYSQL_DNS
+
+
+__all__ = ('db',)
+
+
+class DB:
+    """
+    Handles db connections
+    """
+    def __init__(self):
+        self.connection_pool: Pool = None
+        self.testing_conn = None
```

Comment:
Separate code just for testing is a code smell. The underlying issue here is that DB has tight coupling to its configuration (MYSQL_DNS), and it's a singleton so it can't be reinitialized. The correct solution here would be:
* Instantiate DB in some parent object and pass it around instead of using a singleton,
* Pass DB configuration explicitly in the constructor.

Even ignoring code smell, the huge issue here is that we *don't test actual production code* - only code we wrote just for the test.

Reply:
If you passed the connection pool as an argument in the constructor, you'd be able to replace it for testing without any extra code. That's the whole point of dependency injection. I solved the exact same problem in my server, you can look it up.